backend/backends/datadog/datadog.go

```diff
@@ -105,6 +105,22 @@ func (d *client) SendMetrics(ctx context.Context, metrics *types.MetricMap) erro
 	if metrics.NumStats == 0 {
 		return nil
 	}
+	return d.postMetrics(d.prepareSeries(metrics))
+}
+
+// SendMetricsAsync flushes the metrics to Datadog, preparing payload synchronously but doing the send asynchronously.
+func (d *client) SendMetricsAsync(ctx context.Context, metrics *types.MetricMap, c backendTypes.SendCallback) {
+	if metrics.NumStats == 0 {
+		c(nil)
+		return
+	}
+	ts := d.prepareSeries(metrics)
+	go func() {
```

Comment:
As per language spec https://golang.org/ref/spec#Go_statements:
```
The function value and parameters are evaluated as usual in the calling goroutine, but unlike with a regular call, program execution does not wait for the invoked function to complete.
```
I don't want `d.postMetrics(ts)` to be evaluated in the calling goroutine - the point is to execute it in a new goroutine. So it must be wrapped in a func.